### ISSUE TITLE
amount of solute dispensed from shaker exceeds maximum

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/beers-law-lab#176]
The amount of solute dispensed by the shaker should be limited by ConcentrationModel line 33:

```var SOLUTE_AMOUNT_RANGE = new Range( 0, 6, 0 ); // moles```

But in https://github.com/phetsims/beers-law-lab/issues/169#issuecomment-221095734 and elsewhere, it was reported (and can be demonstrated) that the concentration in 1L exceeds 6 moles.  So there's something amiss.

**Comment by pixelzoom** [comment]
The problem is at ShakerParticles line 102:

```js
solution.soluteAmountProperty.set( solution.soluteAmountProperty.get() + ( 1 / solution.soluteProperty.get().particlesPerMole ) );
```

This should be limited to SOLUTE_AMOUNT_RANGE, something like this:

```js
solution.soluteAmountProperty.set( Math.min( ConcentrationModel.SOLUTE_AMOUNT_RANGE.max, solution.soluteAmountProperty.get() + ( 1 / solution.soluteProperty.get().particlesPerMole ) ) );
```

**Comment by pixelzoom** [comment]
Fixed in master and 1.4 branches, will be verified in 1.4.0-rc.2.

**Comment by pixelzoom** [comment]
@phet-steele please verify in this dev version:
http://www.colorado.edu/physics/phet/dev/html/beers-law-lab/1.4.0-dev.8/beers-law-lab_en.html

The easiest way to verify is:
1. start sim
2. go to Concentration screen
3. select "Sodium chloride" solute
4. fill beaker to 1L
5. put concentration probe in solution
6. shake the shaker, until it's empty (which dispenses 6 moles)
7. concentration meter should read 6 mol/L

If all looks well, feel free to close this issue.

**Comment by phet-steele** [comment]
@pixelzoom the solute is currently stopping at 6 moles as intended. However, there is a visual excess of solute dispensed, and in the words of of Chris on skype:

[6/6/16, 11:32:48 AM] Chris Malley: The issue is that the max is computed based on the amount of solute that has hit the surface of the solution, not the amount of solute actually coming out of the shaker.
[6/6/16, 11:33:17 AM] Chris Malley: so there's some excess solute dispensed that just disappears.

**Comment by pixelzoom** [comment]
The current fix is better than it was, but yes, there's a small amount of dispensed solute that simply disappears.  Fixing this will be a major change to the model.  If you want this for 1.4 release, then we start over with that release.  My recommendation is to deploy 1.4 with the current fix, then (possibly) address more completely in a future 1.4 maintenance release or (more likely) a 1.5 release.  @ariel-phet do you agree?

**Comment by ariel-phet** [comment]
@pixelzoom - agreed with your recommendation

**Comment by pixelzoom** [comment]
Remaining issue moved to https://github.com/phetsims/beers-law-lab/issues/179.  Closing.

**Comment by pixelzoom** [comment]
Reopened until verified in next RC.

**Comment by pixelzoom** [comment]
Already verified in https://github.com/phetsims/beers-law-lab/issues/176#issuecomment-224031009

**Comment by pixelzoom** [comment]
Published in Beer's Law Lab 1.4.0 and Concentration 1.3.0.
